image/src/routes/ipfs.ts

```diff
@@ -0,0 +1,151 @@
+import { Hono } from 'hono'
+import { cors } from 'hono/cors'
+import { CACHE_DAY, Env } from '../utils/constants'
+import { fetchIPFS } from '../utils/ipfs'
+import { ipfsToCFI } from '../utils/cloudflare-images'
+import { allowedOrigin } from '../utils/cors'
+import type { CFIApiResponse } from '../utils/types'
+
+const app = new Hono<{ Bindings: Env }>()
+
+app.use('/*', cors({ origin: allowedOrigin }))
+app.get('/*', async (c) => {
+  const { original } = c.req.query()
+  const isOriginal = original === 'true'
+
+  const url = new URL(c.req.url)
+  const path = url.pathname.replace('/ipfs/', '')
+  const fullPath = `${path}${url.search}`
+
+  // TODO: check response from cache
+  // ----------------------------------------
```

Comment:
cc is this still a TODO?

Reply:
Yes, I will investigate in a separate PR. on this one related to this issue: https://github.com/kodadot/workers/pull/162/files#diff-0f42d024452daf51c4b80efbd4f07ce3d5e123da6bc875c6740ff89d9d788903R125